internal/provider/data_source.go

```diff
@@ -70,45 +69,51 @@ func dataSourceRead(d *schema.ResourceData, meta interface{}) error {
 
 	resp, err := client.Do(req)
 	if err != nil {
-		return fmt.Errorf("Error making a request: %s", err)
+		return append(diags, diag.Errorf("Error making request: %s", err)...)
 	}
 
 	defer resp.Body.Close()
 
 	if resp.StatusCode != 200 {
-		return fmt.Errorf("HTTP request error. Response code: %d", resp.StatusCode)
+		return append(diags, diag.Errorf("HTTP request error. Response code: %d", resp.StatusCode)...)
 	}
 
 	contentType := resp.Header.Get("Content-Type")
-	if contentType == "" || isContentTypeAllowed(contentType) == false {
-		return fmt.Errorf("Content-Type is not a text type. Got: %s", contentType)
+	if contentType == "" || isContentTypeText(contentType) == false {
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Warning,
+			Summary:  "Content-Type is not recognized as a text type, got %q",
```

Comment:
```suggestion
			Summary:  fmt.Sprintf("Content-Type is not recognized as a text type, got %q", contentType),
```
However be aware that message seems a little off if `contentType == ""`